### PR TITLE
refactor: separate `CdcScan` from `Scan` in `logical` and `core`

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Part of https://github.com/risingwavelabs/risingwave/issues/13146.
Later we can also rename `Scan` to `RwTableScan` to make the distinction clearer.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/frontend/src/optimizer/plan_node/generic/cdc_scan.rs [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+use risingwave_common::util::column_index_mapping::ColIndexMapping;
+use risingwave_common::util::sort_util::ColumnOrder;
+
+use super::GenericPlanNode;
+use crate::catalog::ColumnId;
+use crate::expr::ExprRewriter;
+use crate::optimizer::optimizer_context::OptimizerContextRef;
+use crate::optimizer::property::FunctionalDependencySet;
+
+/// [`CdcScan`] returns contents of a table or other equivalent object
```
Comment (fuyufjh) [inline src/frontend/src/optimizer/plan_node/generic/cdc_scan.rs:30]:
```suggestion
/// [`CdcScan`] reads rows of a table from external upstream database
```

File: src/frontend/src/optimizer/plan_node/logical_cdc_scan.rs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+use crate::expr::ExprRewriter;
+use crate::optimizer::optimizer_context::OptimizerContextRef;
+use crate::optimizer::plan_node::{
+    ColumnPruningContext, PredicatePushdownContext, RewriteStreamContext, StreamCdcTableScan,
+    ToStreamContext,
+};
+use crate::optimizer::property::Order;
+use crate::utils::{ColIndexMapping, Condition};
+
+/// `LogicalCdcScan` returns contents of a table or other equivalent object
```
Comment (fuyufjh) [inline src/frontend/src/optimizer/plan_node/logical_cdc_scan.rs:37]:
```suggestion
/// `LogicalCdcScan` reads rows of a table from external upstream database
```

**Review by fuyufjh — APPROVED** [review]
LGTM

**Comment by kwannoel** [comment]
Let @StrikeW take a look first.

**Pull request merged**